src/components/panel/panel.js

```diff
@@ -0,0 +1,284 @@
+ *     panel. Defaults to false.
+ *   - `clickOutsideToClose` - `{boolean=}`: Whether the user can click outside the
+ *     panel to close it. Defaults to false.
+ *   - `disableParentScroll` - `{boolean=}`: Whether the user can scroll the page
+ *     behind the panel. Defaults to false.
+ *   - `fullScreen` - `{boolean=}`: Whether the panel should be full screen. Applies
+ *     the class `.md-panel-fullscreen` to the panel on open. Defaults to false.
+ *   - `trapFocus` - `{boolean=}`: Whether focus should be trapped within the panel.
+ *     If `trapFocus` is true, the user will not be able to interact with the rest of
+ *     the page until the panel is dismissed. Defaults to false.
+ *   - `focusOnOpen` - `{boolean=}`: An option to override focus behavior on open.
+ *     Only disable if focusing some other way, as focus management is required for
+ *     panels to be accessible. Defaults to true.
+ *
+ * @returns {$mdPanelRef} panelRef
+ */
+
+
+
+/**
+ * A service that is used for controlling/displaying panels on the screen.
+ *
+ * @param $rootElement
+ * @param $injector
+ * @final
+ * @constructor
+ * @ngInject
+ */
+function MdPanelService($rootElement, $injector) {
+  // Default config options for the panel.
+  this.defaultConfigOptions_ = {
+    parent: $rootElement
+  };
+
+  this.config_ = this.defaultConfigOptions_;
+  this.injector_ = $injector;
+}
+
+
+MdPanelService.prototype.create = function(opt_config) {
+  var configSettings = opt_config || {};
+  for (var key in configSettings) {
+    if (configSettings.hasOwnProperty(key)) {
+      this.config_[key] = configSettings[key];
+    }
+  }
+
+  return new $mdPanelRef(this.injector_, this.config_);
+}
+
+
+/**
+ * @ngdoc object
+ * @name $mdPanelRef
+ *
+ * @description
+ * A reference to a created panel. This reference contains a unique id for the panel,
+ * along with the following properties:
+ *   - `id` - `{string}: The unique id for the panel. This id is used to track when
+ *     a panel was interacted with.
+ *   - `config` - `{Object=}`: The entire config object that was used in create.
+ *
+ * TODO(ErinCoughlan): Add the following properties.
+ *   - `isOpen` - `{boolean}`: Whether the panel is attached to the DOM. Visibility
+ *     to the user does not factor into isOpen.
+ *   - `isHidden` - `{boolean}`: Whether the panel is attached to the DOM, but not
+ *     visible to the user.
+ *   - `onDomAdded` - `{function=}`: Callback function used to announce when the
+ *     panel is added to the DOM.
+ *   - `onOpenComplete` - `{function=}`: Callback function used to announce when the
+ *     open() action is finished.
+ *   - `onRemoving` - `{function=}`: Callback function used to announce the
+ *     close/hide() action is starting. This allows developers to run custom
+ *     animations in parallel the close animations.
+ *
+ * @param $injector
+ * @param config
+ *
+ */
+function $mdPanelRef($injector, config) {
```

Comment:
@gmoothart @KarenParker @DerekLouie I've moved all the public documentation to the top of the file. When we add new public methods, we need to update the docs there, but within the implementation section, we can follow our existing coding style.